app/com/arpnetworking/metrics/portal/alerts/impl/DailyPartitionCreator.java

```diff
@@ -0,0 +1,283 @@
+            .log();
+        final Optional<Exception> executeResult = execute();
+        getTimers().startPeriodicTimer(TICKER_NAME, TICK, TICK_INTERVAL);
+        getSender().tell(executeResult, getSelf());
+    }
+
+    private void tick() {
+        recordCounter("tick", 1);
+
+        final Instant now = _clock.instant();
+        if (_schedule.nextRun(_lastRun).map(run -> run.isBefore(now)).orElse(true)) {
+            getSelf().tell(EXECUTE, getSelf());
+        }
+    }
+
+    // Wrapper to propagate any errors that occurred to the caller.
+    // This is really only useful on a call to `start`.
+    private Optional<Exception> execute() {
+        final LocalDate startDate = ZonedDateTime.ofInstant(_clock.instant(), _clock.getZone()).toLocalDate();
+        final LocalDate endDate = startDate.plusDays(_lookahead);
+
+        LOGGER.info().setMessage("Creating daily partitions for table")
+                .addData("schema", _schema)
+                .addData("table", _table)
+                .addData("startDate", startDate)
+                .addData("endDate", endDate)
+                .log();
+
+        Optional<Exception> error = Optional.empty();
+        try {
+            execute(_schema, _table, startDate, endDate);
+            _lastRun = Optional.of(_clock.instant());
+        } catch (final PersistenceException e) {
+            error = Optional.of(e);
+            LOGGER.error().setMessage("Failed to create daily partitions for table")
+                    .addData("schema", _schema)
+                    .addData("table", _table)
+                    .addData("startDate", startDate)
+                    .addData("endDate", endDate)
+                    .setThrowable(e)
+                    .log();
+        }
+        recordCounter("create", error.isPresent() ? 0 : 1);
+        return error;
+    }
+
+    /**
+     * Create a series of daily partitions for the given parameters.
+     *
+     * @param schema the database schema
+     * @param table the parent table
+     * @param startDate the start date, inclusive.
+     * @param endDate the end date, exclusive.
+     */
+    protected void execute(final String schema,
+                         final String table,
+                         final LocalDate startDate,
+                         final LocalDate endDate) {
+        // While this query does not return anything meaningful semantically,
+        // it still returns a "non-empty" void result and so we can't use the
+        // ordinarily more appropriate SqlUpdate type.
+        final SqlQuery sql = _ebeanServer.createSqlQuery(
+                "select * from create_daily_partition(?::text, ?::text, ?::date, ?::date)")
+                .setParameter(1, schema)
+                .setParameter(2, table)
+                .setParameter(3, startDate)
+                .setParameter(4, endDate);
+
+        sql.findOneOrEmpty().orElseThrow(() -> new IllegalStateException("Expected a single empty result."));
```

Comment:
This is a top level message handler. Throwing the exception here will just result in an uncaught message handler exception in Akka. Seems like we could do better handling, logging and instrumenting this failure. Thoughts?

Reply:
This is not the top-level handler - the actual handler is the `execute` overload with no parameters. This has the actual DB logic but the other code adds logging/metrics.

It is the case though that the wrapper code will not handle this illegal state exception which would still cause the problem you're mentioning albeit for different reasons so I could probably handle that.